### PR TITLE
[Merged by Bors] - Added deployment type

### PR DESCRIPTION
**Pull request by simlay**

### CODE REVIEW

File: src/k8-types/src/app/deployment.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    pub message: Option<String>,
+    pub reason: Option<String>,
+    pub status: String,
+    pub type_: String,
+}
+
+impl Status for DeploymentStatus {}
+
+/*
+#[cfg(test)]
```
Comment (simlay) [inline src/k8-types/src/app/deployment.rs:84]: I copied this from the stateful sets module. I suppose we should delete or fix them.

File: src/k8-types/src/app/deployment.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        todo!();
+    }
+}
+
+#[derive(Deserialize, Serialize, Default, Debug, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct DeploymentStatus {
+    pub available_replicas: Option<i32>,
+    pub collision_count: Option<i32>,
+    pub conditions: Option<Vec<DeploymentCondition>>,
```
Comment (sehz) [inline src/k8-types/src/app/deployment.rs:62]: Don't need Option for Vec.  Use `default` serde attribute

File: src/k8-types/src/app/deployment.rs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+#[derive(Deserialize, Serialize, Default, Debug, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct DeploymentCondition {
+    pub last_transition_time: Option<String>,
+    pub last_update_time: Option<String>,
+    pub message: Option<String>,
+    pub reason: Option<String>,
+    pub status: String,
+    pub type_: String,
```
Comment (sehz) [inline src/k8-types/src/app/deployment.rs:78]: #[serde(rename = "type")]

File: src/k8-types/src/app/deployment.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    pub selector: LabelSelector,
+    pub strategy: Option<DeploymentStrategy>,
+    pub template: TemplateSpec<PodSpec>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone, PartialEq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct DeploymentStrategy {
+    pub rolling_update: Option<RollingUpdateDeployment>,
+    pub type_: Option<String>,
```
Comment (sehz) [inline src/k8-types/src/app/deployment.rs:34]: #[serde(rename = "type")]

**Review by sehz — CHANGES_REQUESTED** [review]

File: src/k8-types/src/app/deployment.rs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+impl Spec for DeploymentSpec {
+    type Status = DeploymentStatus;
+    type Header = DefaultHeader;
+
+    fn metadata() -> &'static Crd {
+        &DEPLOYMENT_API
+    }
+
+    fn make_same(&mut self, _other: &Self) {
+        todo!();
```
Comment (simlay) [inline src/k8-types/src/app/deployment.rs:54]: @sehz I'm not sure what to do here.

Comment (sehz) [reply src/k8-types/src/app/deployment.rs:54]: This is optional where you need custom logic to deal with k8 strategic merge.  Just remove it

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**